### ISSUE TITLE
java 8 error

### ISSUE THREAD

**Issue by kkarwows** [adamw/veripacks#4]
I have error under java 8:
```
java.lang.IllegalArgumentException: INVOKESPECIAL/STATIC on interfaces require ASM 5
	at org.objectweb.asm.MethodVisitor.visitMethodInsn(Unknown Source)
	at org.objectweb.asm.ClassReader.a(Unknown Source)
	at org.objectweb.asm.ClassReader.b(Unknown Source)
	at org.objectweb.asm.ClassReader.accept(Unknown Source)
	at org.objectweb.asm.ClassReader.accept(Unknown Source)
	at org.veripacks.reader.dependencies.ClassDependenciesReader.read(ClassDependenciesReader.scala:25)
	at org.veripacks.reader.MetadataReader$$anonfun$1.apply(MetadataReader.scala:30)
	at org.veripacks.reader.MetadataReader$$anonfun$1.apply(MetadataReader.scala:15)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:252)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:252)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:252)
	at scala.collection.immutable.List.flatMap(List.scala:344)
	at org.veripacks.reader.MetadataReader.readUsagesAndAccessDefinitions(MetadataReader.scala:15)
	at org.veripacks.Veripacks.verify(Veripacks.scala:20)
	at org.veripacks.Veripacks.verify(Veripacks.scala:7)
```
My java version:
java version "1.8.0_45"
Java(TM) SE Runtime Environment (build 1.8.0_45-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.45-b02, mixed mode)

This error appear recently (earlier everything worked ok).

**Comment by adamw** [comment]
Can you check your classpath and see what version of ASM you have? Is it v4 or v5? If it's v4, maybe you could try the version mentioned in #3?

**Comment by kkarwows** [comment]
Changing verifier version to 0.4.2-asm4-SNAPSHOT didn't help:
```
java.lang.NoClassDefFoundError: org/veripacks/VeripacksBuilder$
	at **********.VeripacksTest.veripacksTest(VeripacksTest.java:10)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:78)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:212)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:68)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)
Caused by: java.lang.ClassNotFoundException: org.veripacks.VeripacksBuilder$
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 27 more
```

**Comment by kkarwows** [comment]
I think that problem is caused by ```Opcodes.ASM4``` in https://github.com/adamw/veripacks/blob/release-0.4.2/verifier/src/main/scala/org/veripacks/reader/dependencies/DependencyVisitor.scala#L163 and other occurence.

**Comment by adamw** [comment]
I just published 0.4.2-SNAPSHOT which uses `Opcodes.ASM5` - could you check?

**Comment by kkarwows** [comment]
Now i have different error:
```
java.lang.IllegalArgumentException: null
	at org.objectweb.asm.ClassReader.<init>(Unknown Source)
	at org.objectweb.asm.ClassReader.<init>(Unknown Source)
	at org.objectweb.asm.ClassReader.<init>(Unknown Source)
	at org.veripacks.reader.ClassReaderProducer$.create(ClassReaderProducer.scala:11)
	at org.veripacks.reader.MetadataReader$$anonfun$1.apply(MetadataReader.scala:16)
	at org.veripacks.reader.MetadataReader$$anonfun$1.apply(MetadataReader.scala:15)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:252)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:252)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:252)
	at scala.collection.immutable.List.flatMap(List.scala:344)
	at org.veripacks.reader.MetadataReader.readUsagesAndAccessDefinitions(MetadataReader.scala:15)
	at org.veripacks.Veripacks.verify(Veripacks.scala:20)
	at org.veripacks.Veripacks.verify(Veripacks.scala:7)
	at *******.VeripacksTest.veripacksTest(VeripacksTest.java:13)
```

**Comment by kkarwows** [comment]
Exception is thrown from: org.objectweb.asm.ClassReader:
```
    public ClassReader(final byte[] b, final int off, final int len) {
        this.b = b;
        // checks the class version
        if (readShort(off + 6) > Opcodes.V1_7) {
            throw new IllegalArgumentException();
        }
```

**Comment by kkarwows** [comment]
Can't find 0.4.2-SNAPSHOT in maven repos (https://oss.sonatype.org/content/repositories/snapshots/org/veripacks/veripacks-verifier_2.11/).

**Comment by adamw** [comment]
It's there now, I published yesterday but no idea why it didn't come up - sorry

**Comment by kkarwows** [comment]
With version 0.4.2-SNAPSHOT everythink works fine on java 8!

**Comment by adamw** [comment]
Great, I just released 0.4.3, should be in central in an hour or so

**Comment by kkarwows** [comment]
Thanks!
